txnsync/txngroups.go

```diff
@@ -17,23 +17,524 @@
 package txnsync
 
 import (
+	"bytes"
+	"fmt"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/crypto/compactcert"
+	"github.com/algorand/go-algorand/data/basics"
 	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/logging"
 	"github.com/algorand/go-algorand/protocol"
 )
 
-const maxEncodedTransactionGroup = 10000
-const maxEncodedTransactionGroupEntries = 10000
+const maxEncodedTransactionGroup = 30000
+const maxEncodedTransactionGroupEntries = 30000
+const maxBitmaskSize = (maxEncodedTransactionGroupEntries+7)/8 + 1
+const signatureSize = 64
+const maxSignatureBytes = maxEncodedTransactionGroupEntries * signatureSize
+const addressSize = 32
+const maxAddressBytes = maxEncodedTransactionGroupEntries * addressSize
+
+var errDataMissing = fmt.Errorf("failed to decode: data missing")
```

Comment:
nit: avoid using `fmt.Errorf` unless you're formatting an error. For a fixed-string error, use `errors.New`